terraform/monitoring/03_services.tf

```diff
@@ -0,0 +1,88 @@
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Define specifics for each of the services that will receive SLOs through a custom service
+# The data members needed are to specify the custom service, goals for each SLO (availability and latency),
+# and maximum error budget burn rates.
+variable "custom_services" {
+  type = list(object({
+    service_name = string,
+    service_id = string,
+    availability_goal = number,
+    availability_burn_rate = number,
+    latency_goal = number,
+    latency_threshold = number,
+    latency_burn_rate = number
+  }))
+  default = [
+    {
+      service_name = "Frontend Service"
+      service_id = "frontend"
+      availability_goal = 0.9		# configurable goal for the availability SLO (0.9 = 90% of requests are successsful)
```

Comment:
perhaps worth adding a note about why we set availability goal of only 1 nine. Typically it's 2 nines (99) as minimum, 99.9 as stretch goal and 99.999 as north star for extremely highly available services (not applicable to our demo)